### PR TITLE
Add support for proxmox web interface

### PR DESCRIPTION
**Pull request by fulljackz**
- parser check for wrong username and password
  - scenario is "default" (such as sshd)
  - new collection for proxmox products (others products exist : backup, mail gateway)

Signed-off-by: Unam LN <manu+github@lacavernedemanu.fr>

### CODE REVIEW

**Comment by buixor** [comment]
Hello,

Thanks for the PR, looks great at first glance.

Can you add the `parser.assert` and `scenario.assert` to your PR ? Those are going to be used by tests to detect regressions etc. (`cscli hubtests` should generate - most of - those and displays them on stdout)

Cheers,

**Comment by fulljackz** [comment]
> Hello,
> 
> Thanks for the PR, looks great at first glance.
> 
> Can you add the `parser.assert` and `scenario.assert` to your PR ? Those are going to be used by tests to detect regressions etc. (`cscli hubtests` should generate - most of - those and displays them on stdout)
> 
> Cheers,

Should be ok, as I just discovered crowdsec I'm not sure about parser.assert and sceario.assert content.
Do not hesitate to tell if I'm wrong :smile:

**Comment by buixor** [comment]
Thanks, I'll review :+1:

**Comment by buixor** [comment]
Hey,

Sorry for the delay, I'll try and review it today :smile:

File: parsers/s01-parse/fulljackz/proxmox-logs.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#debug: true
+name: fulljackz/proxmox-logs
+description: "Parse proxmox logs for bruteforce attempts"
+filter: "evt.Parsed.program == 'pvedaemon'"
+onsuccess: next_stage
+pattern_syntax:
+  PVE_AUTH_FAIL: 'authentication failure; rhost=%{IPV6:client_ip6}:%{IPV4:client_ip4} user=%{USERNAME:source_user}@%{WORD:realm} msg=Authentication failure'
+  PVE_NOSU_USER: 'authentication failure; rhost=%{IPV6:client_ip6}:%{IPV4:client_ip4} user=%{USERNAME:source_user}@%{WORD:realm} msg=no such user'
```
Comment (buixor) [inline parsers/s01-parse/fulljackz/proxmox-logs.yaml:8]: did you had the chance to try it with "real" ipv4/ipv6 ? This format looks like you're accessing ipv6 address from an ipv4 address. [`%{IP}`](https://doc.crowdsec.net/docs/next/parsers/patterns#ip) might do the trick ! Please let me know

File: parsers/s01-parse/fulljackz/proxmox-logs.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      apply_on: message
+      statics:
+        - meta: log_type
+          value: pve_nosuch-user
+        - meta: source_user
+          expression: "evt.Parsed.source_user"
+statics:
+    - meta: service
+      value: pvedaemon
+    - meta: source_ip
```
Comment (buixor) [inline parsers/s01-parse/fulljackz/proxmox-logs.yaml:29]: see the comment above, I'm afraid of the impact w/ real ipv6

File: scenarios/fulljackz/proxmox-bf.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+blackhole: 1m
+reprocess: true
+labels:
+ service: pvedaemon
+ type: bruteforce
+ remediation: true
+---
+# Proxmox bad user
+type: leaky
+name: fulljackz/proxmox-bf
```
Comment (buixor) [inline scenarios/fulljackz/proxmox-bf.yaml:18]: can you name it differently, like `proxmox-bf-user-enum`  ?

**Review by buixor — CHANGES_REQUESTED** [review]
Overall LGTM, minor changes :)

File: parsers/s01-parse/fulljackz/proxmox-logs.yaml (hunk 1 above)
Comment (fulljackz) [reply parsers/s01-parse/fulljackz/proxmox-logs.yaml:8]: I replaced `%{IPV6:client_ip6}:%{IPV4:client_ip4}` with `%{IP:client_ip}` and it seems ok.
I tried to connect using ipv6 and added one into log for testing.

So now I have `results["s01-parse"]["fulljackz/proxmox-logs"][12].Evt.Meta["source_ip"] == "::ffff:172.21.10.2"` 
Is this source_ip format is ok and well known by the bouncer ?

**Comment by sabban** [comment]
I'll have a look in this rather soon.

**Comment by fulljackz** [comment]
> I'll have a look in this rather soon.

Following the changes you made in [pureftpd](https://github.com/crowdsecurity/hub/pull/337) I think I'll have to do some changes here to respect your best practices in tests.

**Comment by sabban** [comment]
I am working on this now, but I still have an issue. So I'll get in touch with you as soon I have it figured out.

**Comment by sabban** [comment]
Please review and merge https://github.com/fulljackz/hub/pull/2, then we'll merge it there :)

Thanks again for your contribution !

**Pull request merged**